File: tools/009-clients/follow.go
// net/http — 既定でリダイレクトを追う。
// Go 1.8 以降、ドメインが変わると Authorization などの機微なヘッダを落とす。
package main

import (
	"fmt"
	"io"
	"net/http"
	"os"
	"strings"
	"time"
)

func main() {
	c := &http.Client{Timeout: 20 * time.Second}
	req, _ := http.NewRequest("GET", os.Args[1], nil)
	req.Header.Set("Authorization", "Bearer MEASUREMENT-TOKEN")
	res, err := c.Do(req)
	if err != nil {
		fmt.Println(`{"auth":"error"}`)
		return
	}
	defer res.Body.Close()
	b, _ := io.ReadAll(res.Body)
	fmt.Println(strings.TrimSpace(string(b)))
}

File: tools/009-clients/follow.rb
# ruby — Net::HTTP はリダイレクトを自分では追わない。
# 標準ライブラリで追う道は open-uri なので、そちらで測る。
require "open-uri"
begin
  body = URI.open(ARGV[0], "Authorization" => "Bearer MEASUREMENT-TOKEN", :read_timeout => 20).read
  puts body.strip
rescue => e
  puts %({"auth":"error","detail":"#{e.class}"})
end
